### PR TITLE
tests: Remove old feature extension enabling

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Hopefully the last PR I will have to do this for

### CODE REVIEW

File: tests/unit/atomics.cpp [hunk 1]
```diff
@@ -430,21 +414,12 @@ TEST_F(NegativeAtomic, ImageInt64Drawtime32) {
...
-    } else if (!features2.features.sparseBinding) {
-        GTEST_SKIP() << "sparseBinding feature not supported";
-    } else if (!image_atomic_int64_features.shaderImageInt64Atomics) {
-        GTEST_SKIP() << "shaderImageInt64Atomics feature not supported";
-    }
-    image_atomic_int64_features.sparseImageInt64Atomics = VK_FALSE;  // turn off
-    RETURN_IF_SKIP(InitState(nullptr, &features2));
+    AddRequiredFeature(vkt::Feature::shaderInt64);
+    AddRequiredFeature(vkt::Feature::sparseBinding);
+    AddRequiredFeature(vkt::Feature::sparseResidencyImage2D);
```
Comment (ziga-lunarg) [inline tests/unit/atomics.cpp:420]: `sparseResidencyImage2D` was not required before, is this an intended change?

File: tests/unit/atomics.cpp [hunk 2]
```diff
@@ -1188,20 +1152,13 @@ TEST_F(NegativeAtomic, Float2WidthMismatch) {
...
-        GTEST_SKIP() << "Required float 16 atomic features not supported";
-    }
-    // turn off 32-bit support
-    atomic_float2_features.shaderBufferFloat32AtomicMinMax = VK_FALSE;
-    RETURN_IF_SKIP(InitState(nullptr, &storage_16_bit_features));
+
+    AddRequiredFeature(vkt::Feature::shaderFloat16);
+    AddRequiredFeature(vkt::Feature::storageBuffer16BitAccess);
+    AddRequiredFeature(vkt::Feature::shaderBufferFloat16AtomicMinMax);
+    AddRequiredFeature(vkt::Feature::shaderSharedFloat32AtomicMinMax);
```
Comment (ziga-lunarg) [inline tests/unit/atomics.cpp:1159]: `shaderSharedFloat32AtomicMinMax` was not required before, is this intended?

File: tests/unit/atomics.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply tests/unit/atomics.cpp:420]: yes, it was just implicitly being added and needed for this test

**Review by arno-lunarg — APPROVED** [review]

File: tests/unit/atomics.cpp (hunk 2 above)
Comment (arno-lunarg) [reply tests/unit/atomics.cpp:1159]: looking at the test's compute shaders, looks like it is intended

**Pull request merged**